psl-clojure/template/resources/leiningen/new/psl_clojure/core.clj

```diff
@@ -0,0 +1,171 @@
+(ns {{namespace}}
+  "Example PSL Clojure program."
+  (:require
+   [incanter.core :as in]
+   [incanter.io :as ini]
+   [clojure.java.io :as io]
+   [psl-clojure.core :as psl]
+   [clj-util.core :as u]
+   )
+  (:use
+   [incanter.core]
+   [incanter.io])
+  (:import
+   [org.linqs.psl.model.term ConstantType]
+   [org.linqs.psl.evaluation.statistics ContinuousEvaluator]
+   [org.linqs.psl.database.rdbms.driver H2DatabaseDriver H2DatabaseDriver$Type]
+   ))
+
+;;; Predicates
+
+(defn predicates
+  "Add predicates to the model"
+  [model]
+  (psl/add-predicate model knows
+                     {"types" (repeat 2 ConstantType/UniqueStringID)
+                      "names" ["p1" "p2"]})
+  (psl/add-predicate model lived
+                     {"types" (repeat 2 ConstantType/UniqueStringID)
+                      "names" ["person" "location"]})
+  (psl/add-predicate model likes
+                     {"types" (repeat 2 ConstantType/UniqueStringID)
+                      "names" ["person" "topic"]})
+  model)
+
+;;; Rules
+
+(defn rules
+  "Add rules to the model"
+  [model datastore]
+  (psl/add-rule model
+                (psl/IMPL
+                 (psl/AND
+                  (lived 'P1 'L)
+                  (lived 'P2 'L)
+                  (psl/NEQ 'P1 'P2))
+                 (knows 'P1 'P2))
+                20 true "nearby-people-known")
+  (psl/add-rule model
+                (psl/IMPL
+                 (psl/AND
+                  (lived 'P1 'L1)
+                  (lived 'P2 'L2)
+                  (psl/NEQ 'P1 'P2)
+                  (psl/NEQ 'L1 'L2))
+                 (psl/NOT (knows 'P1 'P2)))
+                5 true "distant-people-unknown")
+  (psl/add-rule model
+                (psl/IMPL
+                 (psl/AND
+                  (likes 'P1 'T)
+                  (likes 'P2 'T)
+                  (psl/NEQ 'P1 'P2))
+                 (knows 'P1 'P2))
+                10 true "homophily")
+  (psl/add-rule model
+                (psl/IMPL
+                 (psl/AND
+                  (knows 'P1 'P2)
+                  (knows 'P2 'P3)
+                  (psl/NEQ 'P1 'P3))
+                 (knows 'P1 'P3))
+                5 true "transitive")
+  (psl/add-rule-string model datastore "knows(P1, P2) = knows(P2, P1) .")
+  (psl/add-rule model
+                (psl/NOT (knows 'P1 'P2))
+                5 true "prior")
+  model)
+
+;;; Handy functions --- modify to suit
```

Comment:
Calling these "Handy functions" seems to trivialize them a bit.
These are core functions that make everything work.